src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcWebSocketsModule.cs

```diff
@@ -43,21 +45,26 @@ public class JsonRpcWebSocketsModule : IWebSocketsModule
             IJsonRpcService jsonRpcService,
             IJsonRpcLocalStats jsonRpcLocalStats,
             ILogManager logManager,
-            IJsonSerializer jsonSerializer)
+            IJsonSerializer jsonSerializer,
+            IJsonRpcUrlCollection jsonRpcUrlCollection)
         {
             _jsonRpcProcessor = jsonRpcProcessor;
             _jsonRpcService = jsonRpcService;
             _jsonRpcLocalStats = jsonRpcLocalStats;
             _logManager = logManager;
             _jsonSerializer = jsonSerializer;
+            _jsonRpcUrlCollection = jsonRpcUrlCollection;
         }
 
-        public ISocketsClient CreateClient(WebSocket webSocket, string clientName)
+        public ISocketsClient CreateClient(WebSocket webSocket, string clientName, int port)
         {
+            JsonRpcUrl jsonRpcUrl = _jsonRpcUrlCollection.FirstOrDefault(x => x.Port == port && x.RpcEndpoint.HasFlag(RpcEndpoint.WebSocket));
```

Comment:
I'm not sure if I like how we configure these additional URLs and then actually we use only ports.
Maybe the config should contain only ports instead of scheme+host+port ?
@MarekM25 ?

Reply:
> I'm not sure if I like how we configure these additional URLs and then actually we use only ports. Maybe the config should contain only ports instead of scheme+host+port ? @MarekM25 ?

No, it shouldn't contain only ports. We want to be able to set up the engine module for localhost only. However, other RPC modules could be public. That is the goal of this issue.

Reply:
Right, since the urls are pass in to UseUrls() in JsonRpcRunner.cs WebHostBuilder.

Reply:
also potentially slow please use dictionary